foam_controlwrapper/foam_mesh.py

```diff
@@ -0,0 +1,674 @@
+                    raise NotImplementedError(error_str)
+
+            return cell
+
+        except KeyError:
+            error_str = "Trying to get an non-existing cell with uuid: {}"
+            raise ValueError(error_str.format(uuid))
+
+    def add_point(self, point):
+        message = 'Point addition not supported yet'
+        raise NotImplementedError(message)
+
+    def add_edge(self, edge):
+        message = 'Edge addition not supported yet'
+        raise NotImplementedError(message)
+
+    def add_face(self, face):
+        message = 'Face addition not supported yet'
+        raise NotImplementedError(message)
+
+    def add_cell(self, cell):
+        message = 'Cell addition not supported yet'
+        raise NotImplementedError(message)
+
+    def update_point(self, point):
+        message = 'Point update not supported yet'
+        raise NotImplementedError(message)
+
+    def update_edge(self, edge):
+        message = 'Edge update not supported yet'
+        raise NotImplementedError(message)
+
+    def update_face(self, face):
+        message = 'Face update not supported yet'
+        raise NotImplementedError(message)
+
+    def update_cell(self, cell):
+        """ Updates the information of a cell.
+
+        Gets the mesh cell identified by the same
+        uuid as the provided cell and updates its data
+        with the one provided with the new cell. Cell points
+        are not updated.
+
+        Parameters
+        ----------
+        cell : Cell
+            Cell to be updated
+
+        Raises
+        ------
+        KeyError
+            If the cell was not found in the mesh
+
+        TypeError
+            If the object provided is not a cell
+
+        """
+
+        if cell.uid not in self._uuidToFoamLabel:
+            error_str = "Trying to update a non-existing cell with uuid: "\
+                + str(cell.uid)
+            raise KeyError(error_str)
+
+        # if points are changed raise warning
+        pointLabels = foamface.getCellPoints(self.name,
+                                             self._uuidToFoamLabel[cell.uid])
+        puids = [self._foamPointLabelToUuid[lbl] for lbl in pointLabels]
+        # this needed while FileMesh can give puid as numpy.string (bug)
+        cell_puids = []
+        for puid in cell.points:
+            if type(puid) is not numpy.string_:
+                cell_puids.append(puid)
+            else:
+                cell_puids.append(uuid.UUID(puid, version=4))
+
+        if set(puids) != set(cell_puids):
+            raise Warning("Cell points can't be updated")
+
+        # test if data exists and if not create
+        # (must be changed in the future,
+        # while not efficient to make this in cell level)
+        foamFiles = FoamFiles()
+        dataNames = foamFiles.get_cell_data_names(self.path,
+                                                  self._time)
+
+        for data in cell.data:
+            if data == CUBA.PRESSURE:
```

Comment:
Looking at the if else code it makes me wonder that one could refactor the implementation as suggested for https://github.com/simphony/simphony-openfoam/pull/10/files#r27655417. But the wayforward is not trivial. I would howewver make a big difference in how easy it is to maintain the implementation when more CUBA attributes are supported. So this shoudl be somthing for the future.